superset/db_engine_specs/ocient.py

```diff
@@ -215,32 +222,55 @@ class OcientEngineSpec(BaseEngineSpec):
 
     @classmethod
     def get_table_names(
-        cls, database: "Database", inspector: Inspector, schema: Optional[str]
+        cls, database: Database, inspector: Inspector, schema: Optional[str]
     ) -> List[str]:
         return sorted(inspector.get_table_names(schema))
 
     @classmethod
     def fetch_data(cls, cursor, lim=None):
         rows = super(OcientEngineSpec, cls).fetch_data(cursor)
+                
+        if len(rows) > 0 and type(rows[0]).__name__ == rows:
+            # Peek at the schema to determine which column values, if any,
+            # require sanitization.
+            columns_to_sanitize: List[PlacedSanitizeFunc] = _find_columns_to_sanitize(cursor)
 
-        if (len(rows)) == 0:
-            # No rows were produced
-            return rows
+            if columns_to_sanitize:
+                # At least 1 column has to be sanitized.
+                for row in rows:
+                    for info in columns_to_sanitize:
+                        # Modify the element in-place.
+                        v = row[info.column_index]
+                        row[info.column_index] = info.sanitize_func(v)
 
-        if type(rows[0]).__name__ != 'Row':
-            # TODO what else is returned here?
-            return rows
+        # # We are done with this cursor so we can safely remove it from the cache
+        query_ids_to_delete = [q_id for q_id in cursor_cache if cursor_cache[q_id] == cursor]
```

Comment:
Are we not able to get the `Query` object here? Is there a commonly shared value between the pub-sub?

Reply:
The linear search through a dict here is a code smell, shouldn't this be a cache lookup?

Reply:
No, Query objects simply reference the backend table, and the hashing algorithm for the dict may change, while the underlying data may not. I tried using the objects themselves, and it did not work.

Reply:
I think it's highly unlikely that the python defaultdict implementation would use a hashing function that wasn't idempotent. There has to be _some_ common thread between pub-sub points?

Reply:
The memory addresses change. The link is the id attribute